### PR TITLE
fix menu navigation problem

### PR DESCRIPTION
**Pull request by ez-alexfrazer**
## What did we change?

- Fixes the user menu to go on top of utility links

## Why are we doing this?

Fixes: https://github.com/ezcater/recipe/issues/708

Previously, it was layering strangely on top of the utility links, without a clear background, making it difficult to read 

## Screenshot(s) / Gif(s):

image

## Checklist

- [x] Provide test coverage for the changes made
- [ ] Create a changeset (`npm run changeset`) with a summary of the changes

[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

File: src/components/EzNavigation/EzNavigation.tsx [hunk 1]
```diff
@@ -293,7 +293,12 @@ const EzNavigation: FC<Props> = ({
             )}
             {userMenu && (
               <nav className={menus()} aria-label={t('User menu')}>
-                <UserMenu name={userMenu.name} links={userMenu.links} sidebarToggle={toggle} />
+                <UserMenu
+                  name={userMenu.name}
+                  links={userMenu.links}
+                  isSidebarOpen={!hidden}
```
[the remark this answers is not in the document]
Comment (ez-alexfrazer) [reply src/components/EzNavigation/EzNavigation.tsx:299]: basically, when the sidebar closes, this drawer must also close

File: src/components/EzNavigation/UserMenu.tsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
-type MenuProps = {
-  name: string;
-  links: any;
-  sidebarToggle: (ev: React.SyntheticEvent<any>) => void;
-};
+interface MenuProps {
+  readonly name: string;
+  readonly links: any;
+  readonly isSidebarOpen?: boolean;
+  readonly sidebarToggle: MouseEventHandler<HTMLElement>;
```
[the remark this answers is not in the document]
Comment (ez-alexfrazer) [reply src/components/EzNavigation/UserMenu.tsx:20]: More accurate, the click handler is just passed through

File: src/components/EzNavigation/UserMenu.tsx [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
 import EzLink from '../EzLink';
 import {clsx, wrapEvents} from '../../utils';
 import EzPopover from '../EzPopover';
 
-type MenuProps = {
-  name: string;
-  links: any;
-  sidebarToggle: (ev: React.SyntheticEvent<any>) => void;
-};
+interface MenuProps {
```
[the remark this answers is not in the document]
Comment (ez-alexfrazer) [reply src/components/EzNavigation/UserMenu.tsx:16]: [from the typescript cheatsheet](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example/#tldr)

> Use Interface until You Need Type

Main reason in this specific case:

> always use interface for public API's definition when authoring a library or 3rd party ambient type definitions, as this allows a consumer to extend them via declaration merging if some definitions are missing.

**Review by jbelang — APPROVED** [review]
I'll defer to @danidewitt for final approval.

File: src/components/EzNavigation/UserMenu.tsx [hunk 4]
```diff
@@ -78,57 +89,82 @@ const iconMenuItem = theme.css({
...
+
+const UserMenuTrigger = forwardRef<HTMLButtonElement, TriggerProps>(({name, ...props}, ref) => {
+  return (
+    <button
+      {...props}
+      className={clsx(menuStyles(), iconMenuItem(), buttonReset())}
+      type="button"
+      ref={ref}
+    >
+      <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 32 32">
```
Comment (danidewitt) [inline src/components/EzNavigation/UserMenu.tsx:102]: why not keep this extracted? should we really have it somewhere else (especially for readability)?

**Comment by danidewitt** [comment]
@ez-alexfrazer we're going to need a [changeset](https://ezcater.github.io/recipe/guides/contributing#following-semantic-versioning) for this, based on our [versioning](https://github.com/ezcater/recipe/blob/main/doc-site/src/pages/guides/versioning-and-changelog.md)

**Review by danidewitt — APPROVED** [review]

**Pull request merged**